### DISCUSSION TITLE
How do you sync your notes directories (desktop and mobile)?

### DISCUSSION THREAD

**Discussion by minthemiddle** [notaapp/talk#342] [Questions]
How do you sync your notes directories - both between desktop machines and mobile devices?

I have been using iCloud so I can access and edit the notes on iOS via ia Writer. But I'm not happy at all with it:

- iCloud does not work well with git, my repo has been broken several times already (this does not happen when I have the dir elsewhere)
- iCloud's sync is slow and buggy

What can you recommend?

**Comment by astoilkov** [comment]
Hey, sorry for the delayed answer.

Both me and @erusev use both Dropbox and git. We have a [save hook](https://docs.nota.md/save-hook.html) that automatically creates a daily commit. We've had a few problems with Dropbox but generally, it's working well and we're happy with it.

On mobile, we use the Dropbox app, iA Writer, or the Nota mobile app we've been working on recently.

**Reply by minthemiddle** [reply]
@astoilkov Thanks for sharing! Would you mind sharing your daily save hook? I'm using a per-change commit with a timestamp but I feel it's overengineering and it certainly makes my git repo grow very quickly…

**Reply by minthemiddle** [reply]
@astoilkov On Dropbox and iA Writer: Do you use it on a per-file basis? As per [iA's documentation](https://ia.net/writer/support/general/dropbox), it seemed to be the only possibility to me and I could not browse the library in iA Writer but would need to use Apple's Files for that. With iCloud sync, I can open the full library in iA Writer with full search. I'm asking as I'd like to switch back to Dropbox because of better sync performance but am afraid that the experience is worse on iOS.

**Reply by astoilkov** [reply]
Here is the commit hook:
```sh
#!/bin/bash

# nothing changed, exit
# based on https://stackoverflow.com/a/9393642/200145
# -s: Give the output in the short-format.
# -z: empty string
[[ -z $(git status -s) ]] && exit

# example: 29 June 2022
currentDate=$(date +"%d %B %Y")
# the (author) date of the last commit
commitDate=$(git log -1 --pretty=%ad --date=format:"%d %B %Y")

# add changes
# we need this regardless of how we commit
git add --all

if [[ "$commitDate" == "$currentDate" ]]; then
  # we're still on the same day, so we're still on the same commit
  git commit --amend --no-edit --allow-empty
else
  # it's a new day, so we're on a new commit
  # name the commit with the date
  git commit --allow-empty -m "$currentDate"
fi
```

**Reply by astoilkov** [reply]
I'm on Android with iA Writer where I think the implementation is a bit different. Also, iA Writer is probably the thing that I use the least.

The Dropbox app is nice because you can (on Android) create a shortcut to your notes folder and then search only that folder and open a file.

**Comment by Auratelience** [comment]
> Nota mobile app we've been working on recently.

Not to be pushy, but how far through development are you guys? Are there any screenshots or the like, perhaps?

**Reply by astoilkov** [reply]
We don't have screenshots. We have an app that we have installed on our phones and use it. It's not very stable at the moment.

**Reply by Auratelience** [reply]
Ah ok. Looking forward to it!

**Comment by claudiorrrr** [comment]
Nextcloud for sync here
